.github/workflows/sync_from_airtable.yml

```diff
@@ -127,23 +126,67 @@ jobs:
             const { labels } = JSON.parse(DATA);
             if (labels) {
                 const issue  = JSON.parse(process.env.ISSUE);
-                const ghLabels = issue.data.labels.map((label) => label.name);
-                const keepLabels = []
-                const addLabels = [];
-                const ghLookup = {};
-                for (let j in ghLabels) {
-                    ghLookup[ghLabels[j]] = ghLabels[j];
-                }
-                for (let i in labels) {
-                    if (typeof ghLookup[labels[i]] != 'undefined') {
-                        keepLabels.push(labels[i])
-                        delete ghLookup[labels[i]];
+                const { removeLabels, addLabels } = labels.reduce((acc, labelName, idx) => {
+                    const currentIdx = acc.priorLabels.indexOf(labelName);
+
+                    if (currentIdx === -1) {
+                        acc.addLabels.push(labelName);
                     } else {
-                        addLabels.push(labels[i]);
+                        // Keep the label indices from the priorLabel list
+                        acc.keepLabels[currentIdx] = labelName;
                     }
-                }
-                const removeLabels = Object.keys(ghLookup);
-                if (addLabels && addLabels.length > 0) {
+
+                    if (idx === labels.length - 1) {
+                        // We are in our final loop. Remove empty indices from keepLabels and set those priorLabels for removal.
```

Comment:
` // We are in our final loop.`

Can you use a 'self-documenting variable' to get rid of the comment? 

```js
const inFinalLoop = idx === labels.length - 1;

if (inFinalLoop) {
/* ... */
```

This sort of forces us to update the variable if it's context changes later on vs the code and the comment.